commons/com.b2international.index/src/com/b2international/index/es/admin/EsIndexAdmin.java

```diff
@@ -145,24 +145,23 @@ public void create() {
 		for (DocumentMapping mapping : mappings.getMappings()) {
 			final String index = getTypeIndex(mapping);
 			final String type = mapping.typeAsString();
-			final Map<String, Object> typeMapping = ImmutableMap.of(type,
-					ImmutableMap.builder()
+			final Map<String, Object> typeMapping = ImmutableMap.<String, Object>builder()
 					.put("date_detection", false)
 					.put("numeric_detection", false)
 					.putAll(toProperties(mapping))
-					.build());
+					.build();
 			
 			if (exists(mapping)) {
 				// update mapping if required
 				ImmutableOpenMap<String, MappingMetaData> currentIndexMapping;
 				try {
-					currentIndexMapping = client.indices().getMapping(new GetMappingsRequest().indices(index).types(type)).mappings().get(index);
+					currentIndexMapping = client.indices().getMapping(new GetMappingsRequest().types(type).indices(index)).mappings().get(index);
```

Comment:
Is `types` still needed here? I see that the returned JSON object no longer nests the mapping under a property key.

Reply:
Yes, unfortunately, the GetMapping/PutMapping/CreateIndex action requests still require `type` value to be specified. The new ES HLRC however uses a different set of Request classes which do not require the `type` value. I was able to remove all other `type` fields from the `EsClient` interface and implementations.